logging/logger.go

```diff
@@ -3,40 +3,35 @@ package logging
 
 import (
 	"io"
-	"log"
+	"log/slog"
 	"os"
 )
 
-var logger *log.Logger
+const logFile = "revive.log"
+
+var logger *slog.Logger
 
 // GetLogger retrieves an instance of an application logger which outputs
 // to a file if the debug flag is enabled
-func GetLogger() (*log.Logger, error) {
+func GetLogger() (*slog.Logger, error) {
 	if logger != nil {
 		return logger, nil
 	}
 
-	var writer io.Writer
-	var err error
-
-	writer = io.Discard // by default, suppress all logging output
-	debugModeEnabled := os.Getenv("DEBUG") == "1"
-	if debugModeEnabled {
-		writer, err = os.Create("revive.log")
-		if err != nil {
-			return nil, err
-		}
+	debugModeEnabled := os.Getenv("DEBUG") != ""
+	if !debugModeEnabled {
+		// by default, suppress all logging output
+		return slog.New(slog.NewTextHandler(io.Discard, nil)), nil
```

Comment:
BTW, is there a linter about slog (not necessarily in revive) about replacing `slog.NewTextHandler(io.Discard, nil)` ?

Reply:
I think no, it's not.

Reply:
Maybe we should think about it

https://github.com/search?q=%22slog.NewTextHandler%28io.Discard%2C+nil%29%22&type=code

What would be the best candidate? Revive? Go-critic ?

Reply:
Perhaps it's go-critic

Reply:
Yep, this can be easily done via `ruleguard` rule. Feel free to submit an issue (or even a PR).

Reply:
Thanks for confirmation

It can be done here then:

https://github.com/go-critic/go-critic/blob/master/checkers%2Frules%2Frules.go

Reply:
I consider sloglint as a possible candidate also

https://github.com/go-simpler/sloglint

@tmzane do you think there could be an interest in sloglint to report people using 

`slog.NewTextHandler(io.Discard, nil)` for Go 1.24+, while `slog.DiscardHandler` is available

Apparently, it's massively used

https://github.com/search?q=%22slog.NewTextHandler%28io.Discard%2C+nil%29%22&type=code

If you have interest for this, I will open an issue in your project
If you don't, I will open an issue in go-critric and will code it I think.

Reply:
Hi, I don't mind adding it to sloglint as in slog-focused linter, seems like a useful and easy-to-implement check. Feel free to open an issue, thanks

Reply:
Thanks for creating

- https://github.com/go-simpler/sloglint/issues/78